sockeye/inference.py

```diff
@@ -645,30 +720,31 @@ def translate(self, trans_input: TranslatorInput) -> TranslatorOutput:
                                     score=-np.inf)
 
         if len(trans_input.tokens) > self.max_input_length:
-            logger.debug("Input (%d) exceeds max input length (%d). Splitting into chunks of size %d.",
-                         len(trans_input.tokens), self.buckets_source[-1], self.max_input_length)
+            logger.debug("Input %d has length (%d) that exceeds max input length (%d). Splitting into chunks of size %d.",
+                         trans_input.id, len(trans_input.tokens), self.buckets_source[-1], self.max_input_length)
             token_chunks = utils.chunks(trans_input.tokens, self.max_input_length)
-            translations = [self.translate_nd(*self._get_inference_input(tokens))
+            translations = [self.translate_nd(*self._get_inference_input([tokens]))[0]
                             for tokens in token_chunks]
             translation = self._concat_translations(translations)
-            return self._make_result(trans_input, translation)
         else:
-            return self._make_result(trans_input, self.translate_nd(*self._get_inference_input(trans_input.tokens)))
+            translation = self.translate_nd(*self._get_inference_input([trans_input.tokens]))[0]
+        return self._make_result(trans_input, translation)
 
-    def _get_inference_input(self, tokens: List[str]) -> Tuple[mx.nd.NDArray, int]:
+    def _get_inference_input(self, tokens: List[List[str]]) -> Tuple[mx.nd.NDArray, int]:
         """
-        Returns NDArray of source ids (shape=(1, bucket_key)) and corresponding bucket_key.
+        Returns NDArray of source ids (shape=(num_sent, bucket_key)) and corresponding bucket_key.
 
-        :param tokens: List of input tokens.
+        :param tokens: List of lists of input tokens.
         :return NDArray of source ids and bucket key.
         """
-        bucket_key = data_io.get_bucket(len(tokens), self.buckets_source)
+        bucket_key = data_io.get_bucket(max(len(t) for t in tokens), self.buckets_source)
```

Comment:
t is a bit confusing, one may think this is a single token and you want its length, whereas this is actually the length of the token sequence.
Maybe rename the function argument to sequences or something.